### PR TITLE
Add a constructor that takes RestTemplate instead of field injection

### LINKED ISSUE #4682: Allow RestTemplate configure with DataFlowClientAutoConfiguration
Ideally logic should be extracted into public methods that accept a restTemplate

At least make the dependencies autowired by constructor or @Bean method arguments so we can compose the behavior from outside.

Or a setter for restTemplate

### PR DESCRIPTION
**Pull request by markpollack**
Fixes #4682

### CODE REVIEW

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java [hunk 1]
```diff
@@ -90,6 +89,13 @@ public class DataFlowClientAutoConfiguration {
 	@Autowired
 	private @Nullable OAuth2ClientProperties oauth2ClientProperties;
 
+	public DataFlowClientAutoConfiguration() {
```
Comment (onobc) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java:92]: I am not a fan of mixing constructor and field injection. Wdyt about moving it all to CI such as:
```java
	private DataFlowClientProperties properties;

	private RestTemplate restTemplate;

	private ClientRegistrationRepository clientRegistrations;
	
	private OAuth2AccessTokenResponseClient<OAuth2ClientCredentialsGrantRequest> clientCredentialsTokenResponseClient;

	private OAuth2ClientProperties oauth2ClientProperties;

	public DataFlowClientAutoConfiguration(
			DataFlowClientProperties properties,
			@Nullable RestTemplate restTemplate,
			@Nullable ClientRegistrationRepository clientRegistrations,
			@Nullable OAuth2AccessTokenResponseClient<OAuth2ClientCredentialsGrantRequest> clientCredentialsTokenResponseClient,
			@Nullable OAuth2ClientProperties oauth2ClientProperties) {
		this.properties = properties;
		this.restTemplate = restTemplate;
		this.clientRegistrations = clientRegistrations;
		this.clientCredentialsTokenResponseClient = clientCredentialsTokenResponseClient;
		this.oauth2ClientProperties = oauth2ClientProperties;
	}
```

Comment (klopfdreh) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java:92]: +1 for constructor wiring. Anything I can help with here?

Comment (onobc) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java:92]: Thanks @klopfdreh for the offer. We will get this updated and merged shortly though.

Comment (onobc) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java:92]: We are going to accept this proposal as-is for M2 and then revisit in follow on issue w/ the prospect of leveraging the built-in Spring Boot auto-configured `RestTemplateBuilder`.

Comment (klopfdreh) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java]: Sounds great! Thanks for all the hard work!

**Pull request merged**